### PR TITLE
feat: [FFM-8012]: Refreshing/Back button returns to All Environments

### PR DESCRIPTION
**Pull request by laura-mannus**
### Summary

Once we select the All Environments filter in the dropdown and refresh the page, it should remain on the All Environments view. Also if the user navigates away and then presses back it should do the same. This is due to adding a new parameter to the URL when all environments is selected.

#### Screenshots

https://github.com/harness/harness-core-ui/assets/103498167/8fa76a84-b2a0-4383-8eef-e893e8bd6794

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by crystal-fong — DISMISSED** [review]
looks grand, would this be easily testable?

**Review by Kira-Jennings — DISMISSED** [review]

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx [hunk 1]
```diff
@@ -104,10 +105,18 @@ export const useEnvironmentSelectV2 = ({
...
   useEffect(() => {
+    if (allEnvironments) {
+      const allEnvValue = {
+        label: getString('common.allEnvironments'),
+        value: getString('common.allEnvironments')
+      }
+      setSelectedEnvironment(allEnvValue)
+      refetchProjectFlags()
+      return
+    }
```
Comment (knagurski) [inline src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:116]: Super minor, but we could do with a line break here to break up the wall of code 👍🏻
```suggestion
    }

```

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx [hunk 2]
```diff
@@ -104,10 +105,18 @@ export const useEnvironmentSelectV2 = ({
   }, [preferredEnvironment, selectedEnvironment?.value])
 
   useEffect(() => {
+    if (allEnvironments) {
```
Comment (knagurski) [inline src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:108]: This is missing from the dependencies list

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx (hunk 1 above)
Comment (laura-mannus) [reply src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:116]: cool, thanks have updated

File: src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx (hunk 2 above)
Comment (laura-mannus) [reply src/modules/75-cf/hooks/useEnvironmentSelectV2.tsx:108]: gotcha, have updated

**Review by Kira-Jennings — DISMISSED** [review]

**Review by knagurski — APPROVED** [review]
👍🏻

**Review by crystal-fong — APPROVED** [review]

**Review by knagurski — APPROVED** [review]
👍🏻

**Pull request merged**